spec/index.md

```diff
@@ -496,36 +496,32 @@ limited capabilities.
 
 The list of types is:
 
-* Primitive types:
+* Base types:
     * [Boolean] — `true` or `false`
+    * [Address] - Ethereum address
     * [Numeric] — integer
-* Sequence types:
+* Reference types:
     * [Tuple]
     * [Array]
     * [Bytes]
+    * [Map]
 * User-defined types:
     * [Struct]
     * [Enum]
-    * [Contract]
-    * [Event]
```

Comment:
What's the reason for deleting `Contract` and `Event`?

Reply:
When I removed these, I was thinking that we should reserve the word *type* for data types. However, I'm now thinking this is too limited.

It would probably be a good idea to come up with a solid definition of what a type in Fe is before we classify things as types. So maybe the correct definition would be:

> A type is anything that can be instantiated.

So under that definition, a contract would be a type if we implement contract loading during runtime. Like this:

```
foo = FooContract.at(contract_address)
bar = foo.some_method()
```

So `FooContract.at(contract_address)` would be an expression that returns an instance of the contract type that gets stored it in `foo`.

With events, I'm not quite sure there's a situation where one would be instantiated, so I'm inclined to leave that out of the types spec.

Maybe this is a better classification:

- Data Types
  - Base types:
    - [Boolean] — `true` or `false`
    - [Address] - Ethereum address
    - [Numeric] — integer
  - Reference types:
    - [Tuple]
    - [Array]
    - [Bytes]
    - [Map]
    - [Struct]
    - [Enum]
- Special types:
  - [Function]
  - [Contract]

Reply:
I suppose the current definition is ok.

I'm still not sure if events fits into it though. To me, event definitions exist to simplify writing log statements. You can't really store instances of them anywhere. Also, are you ok with classifying base and reference types under data types?

I agree that contracts should be considered a type.

Reply:
>I'm still not sure if events fits into it though. To me, event definitions exist to simplify writing log statements. You can't really store instances of them anywhere.

That's true but I'm not sure if that should count as a requirement. You could certainly have a shared library holding common event types. I think everything that could appear in a type annotation can be considered a type.

>Also, are you ok with classifying base and reference types under data types?

Yes, I think that classification makes sense :+1: